src/Common/FST.h

```diff
@@ -0,0 +1,128 @@
+#pragma once
```

Comment:
Fixed: more comments were added and FST paper is mentioned in the comment. We also renamed FST class to FiniteStateTransducer to make it clearer, we decided to keep FST as file name because name FiniteStateTransducer is very academic to almost everybody, so FST is more concise(and curious people can dig it out anyway).